pkg/flexvolume/block/driver.go

```diff
@@ -135,27 +134,27 @@ func newClient(config *Config) (client.Interface, error) {
 
 // Init checks that we have the appropriate credentials and metadata API access
 // on driver initialisation.
-func (d OCIFlexvolumeDriver) Init() flexvolume.DriverStatus {
+func (d OCIFlexvolumeDriver) Init(logger *zap.SugaredLogger) flexvolume.DriverStatus {
 	path := GetConfigPath()
 	if d.master {
 		config, err := ConfigFromFile(path)
 		if err != nil {
-			return flexvolume.Fail(err)
+			return flexvolume.Fail(logger, err)
 		}
 		_, err = newClient(config)
 		if err != nil {
-			return flexvolume.Fail(err)
+			return flexvolume.Fail(logger, err)
 		}
 
 		_, err = constructKubeClient()
 		if err != nil {
-			return flexvolume.Fail(err)
+			return flexvolume.Fail(logger, err)
 		}
 	} else {
-		log.Printf("Assuming worker node.")
+		logger.Info("Assuming worker node.")
 	}
 
-	return flexvolume.Succeed()
+	return flexvolume.Succeed(logger)
```

Comment:
I think this is where you want the `zap.New(nil).Sugar()` rather than when calling `Init(zap.New(nil).Sugar())`